src/uct/ib/base/ib_device.c

```diff
@@ -143,6 +143,7 @@ static uct_ib_device_spec_t uct_ib_builtin_device_specs[] = {
   {"BlueField 2", {0x15b3, 0xa2d6},
    UCT_IB_DEVICE_FLAG_MELLANOX | UCT_IB_DEVICE_FLAG_MLX5_PRM |
    UCT_IB_DEVICE_FLAG_DC_V2, 61},
+  {"EFA", {0x1d0f, 0xefa0}, UCT_IB_DEVICE_FLAG_EFA, 0},
```

Comment:
is there a better way to check for EFA?
(we want to deprecate device spec at some point)

Reply:
I think checking the vendor id is the only way to identify a specific device, including EFA.
Is it just the spec array that we want to deprecate? If so, maybe we can query vendor id in `uct_ib_device_init()` and set the UCT_IB_DEVICE_FLAGs for the device accordingly.

Reply:
why not just trying to run uct_ib_efadv_query()?

Reply:
Yeah, I think that works too. Do we want to deprecate UCT_IB_DEVICE_FLAGs too? Later on I use a UCT_IB_DEVICE_FLAG_SRD in addition to UCT_IB_DEVICE_FLAG_EFA.